### PR TITLE
Set repositories' descriptions and URLs

### LINKED ISSUE #20: provide/adjust dandiset stats/description/url for a dandiset upon push
ATM those on https://github.com/dandisets/  have no metadata on them. Hard to impossible to find anything unless know which one you are looking for. Ideally within https://github.com/dandi/dandisets/blob/master/tools/backups2datalad.py we should
- upon initial publish for each repo provide URL which would point to `https://identifiers.org/DANDI:<ID>` e.g. https://identifiers.org/DANDI:000006 .
- upon every push update description so it becomes
  `<files unicode> #files <storage unicode> size,  Contact, Description` to mirror what we have on https://gui.dandiarchive.org/#/dandiset .

- there should be a way to "force trigger" update of descriptions/url (e.g. to populate currently already pushed to github datasets or later if we change what we display there etc)

### PR DESCRIPTION
**Pull request by jwodder**
Closes #20.

The resulting descriptions can be seen on the repositories under https://github.com/dandisets-testing.

### CODE REVIEW

**Review by yarikoptic — CHANGES_REQUESTED** [review]
When I ran it, and it seems it has updated all of them nicely besides then puked on 43 which we do not have yet on github (yet to bring it back on cron ), not sure if worth sweating making that into a "better error" ATM.
<details>
<summary>details</summary> 

```shell
(dandi-cli-dev) dandi@drogon:~/proj/dandi/dandisets-devel$ tools/backups2datalad.py    --backup-remote dandi-dandisets-dropbox     --gh-org dandisets    -J 5 --update-metadata    "$girderdir"    "$ds" 
2020-11-09T16:12:42-0500 [INFO    ] backups2datalad.py Setting metadata for dandisets/000003 ...
2020-11-09T16:12:42-0500 [INFO    ] backups2datalad.py Setting metadata for dandisets/000004 ...
2020-11-09T16:12:43-0500 [INFO    ] backups2datalad.py Setting metadata for dandisets/000005 ...
2020-11-09T16:12:44-0500 [INFO    ] backups2datalad.py Setting metadata for dandisets/000006 ...
2020-11-09T16:12:44-0500 [INFO    ] backups2datalad.py Setting metadata for dandisets/000007 ...
2020-11-09T16:12:45-0500 [INFO    ] backups2datalad.py Setting metadata for dandisets/000008 ...
2020-11-09T16:12:46-0500 [INFO    ] backups2datalad.py Setting metadata for dandisets/000009 ...
2020-11-09T16:12:47-0500 [INFO    ] backups2datalad.py Setting metadata for dandisets/000010 ...
2020-11-09T16:12:47-0500 [INFO    ] backups2datalad.py Setting metadata for dandisets/000011 ...
2020-11-09T16:12:48-0500 [INFO    ] backups2datalad.py Setting metadata for dandisets/000012 ...
2020-11-09T16:12:49-0500 [INFO    ] backups2datalad.py Setting metadata for dandisets/000013 ...
2020-11-09T16:12:49-0500 [INFO    ] backups2datalad.py Setting metadata for dandisets/000015 ...
2020-11-09T16:12:50-0500 [INFO    ] backups2datalad.py Setting metadata for dandisets/000016 ...
2020-11-09T16:12:51-0500 [INFO    ] backups2datalad.py Setting metadata for dandisets/000017 ...
2020-11-09T16:12:51-0500 [INFO    ] backups2datalad.py Setting metadata for dandisets/000018 ...
2020-11-09T16:12:52-0500 [INFO    ] backups2datalad.py Setting metadata for dandisets/000019 ...
2020-11-09T16:12:52-0500 [INFO    ] backups2datalad.py Setting metadata for dandisets/000020 ...
2020-11-09T16:12:55-0500 [INFO    ] backups2datalad.py Setting metadata for dandisets/000021 ...
2020-11-09T16:12:56-0500 [INFO    ] backups2datalad.py Setting metadata for dandisets/000022 ...
2020-11-09T16:12:56-0500 [INFO    ] backups2datalad.py Setting metadata for dandisets/000023 ...
2020-11-09T16:12:57-0500 [INFO    ] backups2datalad.py Setting metadata for dandisets/000024 ...
2020-11-09T16:12:58-0500 [INFO    ] backups2datalad.py Setting metadata for dandisets/000025 ...
2020-11-09T16:12:58-0500 [INFO    ] backups2datalad.py Setting metadata for dandisets/000026 ...
2020-11-09T16:12:59-0500 [INFO    ] backups2datalad.py Setting metadata for dandisets/000027 ...
2020-11-09T16:12:59-0500 [INFO    ] backups2datalad.py Setting metadata for dandisets/000028 ...
2020-11-09T16:13:00-0500 [INFO    ] backups2datalad.py Setting metadata for dandisets/000029 ...
2020-11-09T16:13:01-0500 [INFO    ] backups2datalad.py Setting metadata for dandisets/000030 ...
2020-11-09T16:13:01-0500 [INFO    ] backups2datalad.py Setting metadata for dandisets/000031 ...
2020-11-09T16:13:02-0500 [INFO    ] backups2datalad.py Setting metadata for dandisets/000032 ...
2020-11-09T16:13:03-0500 [INFO    ] backups2datalad.py Setting metadata for dandisets/000033 ...
2020-11-09T16:13:04-0500 [INFO    ] backups2datalad.py Setting metadata for dandisets/000034 ...
2020-11-09T16:13:04-0500 [INFO    ] backups2datalad.py Setting metadata for dandisets/000035 ...
2020-11-09T16:13:05-0500 [INFO    ] backups2datalad.py Setting metadata for dandisets/000036 ...
2020-11-09T16:13:06-0500 [INFO    ] backups2datalad.py Setting metadata for dandisets/000037 ...
2020-11-09T16:13:06-0500 [INFO    ] backups2datalad.py Setting metadata for dandisets/000038 ...
2020-11-09T16:13:07-0500 [INFO    ] backups2datalad.py Setting metadata for dandisets/000039 ...
2020-11-09T16:13:08-0500 [INFO    ] backups2datalad.py Setting metadata for dandisets/000040 ...
2020-11-09T16:13:08-0500 [INFO    ] backups2datalad.py Setting metadata for dandisets/000041 ...
2020-11-09T16:13:09-0500 [INFO    ] backups2datalad.py Setting metadata for dandisets/000042 ...
2020-11-09T16:13:10-0500 [INFO    ] backups2datalad.py Setting metadata for dandisets/000043 ...
Traceback (most recent call last):
  File "tools/backups2datalad.py", line 545, in <module>
    main()
  File "/home/dandi/miniconda3/envs/dandi-cli-dev/lib/python3.8/site-packages/click/core.py", line 829, in __call__
    return self.main(*args, **kwargs)
  File "/home/dandi/miniconda3/envs/dandi-cli-dev/lib/python3.8/site-packages/click/core.py", line 782, in main
    rv = self.invoke(ctx)
  File "/home/dandi/miniconda3/envs/dandi-cli-dev/lib/python3.8/site-packages/click/core.py", line 1066, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/home/dandi/miniconda3/envs/dandi-cli-dev/lib/python3.8/site-packages/click/core.py", line 610, in invoke
    return callback(*args, **kwargs)
  File "tools/backups2datalad.py", line 118, in main
    di.update_metadata(dandisets)
  File "tools/backups2datalad.py", line 383, in update_metadata
    self.gh.get_repo(f"{self.gh_org}/{did}").edit(
  File "/home/dandi/miniconda3/envs/dandi-cli-dev/lib/python3.8/site-packages/github/MainClass.py", line 333, in get_repo
    headers, data = self.__requester.requestJsonAndCheck(
  File "/home/dandi/miniconda3/envs/dandi-cli-dev/lib/python3.8/site-packages/github/Requester.py", line 317, in requestJsonAndCheck
    return self.__check(
  File "/home/dandi/miniconda3/envs/dandi-cli-dev/lib/python3.8/site-packages/github/Requester.py", line 342, in __check
    raise self.__createException(status, responseHeaders, output)
github.GithubException.UnknownObjectException: 404 {"message": "Not Found", "documentation_url": "https://docs.github.com/rest/reference/repos#get-a-repository"}
(dandi-cli-dev) dandi@drogon:~/proj/dandi/dandisets-devel$ echo tools/backups2datalad.py    --backup-remote dandi-dandisets-dropbox     --gh-org dandisets    -J 5 --update-metadata    "$girderdir"    "$ds" 
tools/backups2datalad.py --backup-remote dandi-dandisets-dropbox --gh-org dandisets -J 5 --update-metadata /mnt/backup/dandi/dandiarchive-s3-backup/ /mnt/backup/dandi/dandisets

```
</details>
but left few comments

File: tools/backups2datalad.py [hunk 1]
```diff
@@ -73,6 +80,9 @@
 @click.option(
     "--re-filter", help="Only consider assets matching the given regex", metavar="REGEX"
 )
+@click.option(
+    "--update-metadata", is_flag=True, help="Only update repositories' metadata",
+)
```
[the remark this answers is not in the document]
Comment (jwodder) [reply tools/backups2datalad.py:87]: Renamed.

File: tools/backups2datalad.py [hunk 2]
```diff
@@ -467,5 +518,28 @@ def dandi_logging(dandiset_path: Path):
...
+    '1000000.00 EB'
+    """
+    size = numbytes
+    sizestr = f"{size} B"
+    for prefix in "kMGTPE":
+        if size < 1024:
+            return sizestr
+        size /= 1024
+        sizestr = f"{size:.2f} {prefix}B"
+    return sizestr
```
[the remark this answers is not in the document]
Comment (jwodder) [reply tools/backups2datalad.py]: Done.

**Comment by yarikoptic** [comment]
resolved conflict for `pdb` addition right here in github UI, hopefully I didn't screw up ;)

**Pull request merged**